app/livechat/server/lib/Helper.js

```diff
@@ -122,9 +122,12 @@ export const createLivechatSubscription = (rid, name, guest, agent, department)
 		username: String,
 	}));
 
+	const existingSubscription = Subscriptions.findOneByRoomIdAndUserId(rid, agent.agentId, { fields: { _id: 1 } });
```

Comment:
If the subscription already exists, why do we need to `insertOrUpdate` rather than just return it?

Reply:
I did this to make sure the Subscription contains the latest data. Eg: If the department was changed bcs the chat was forwarded, then we need to change the department within subscription. Hence i've added a `insertOrUpdate` here

Reply:
I got it, but once the chat is forwarded to another department, the nee department is updated on all subscriptions:
`Subscriptions.changeDepartmentByRoomId(rid, newDepartmentId);`

Reply:
Okay got it :+1: If that's the case then we can simply return the subscription and remove the `insertOrUpsert` logic

Reply:
I've made this change, could you please review it again. Thanks :smile: